### ISSUE TITLE
"Large text" - using px rather than pt as unit

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#181]
Something that's been bugging me for a while...

For on-screen/web content, `px` would be a far more natural unit of measure than `pt`. 

`pt` is (generally anyway) derived simply from the reference pixel (where then 1pt ~= 1.333px - see https://drafts.csswg.org/css-values/#absolute-lengths).

having "large text" (https://www.w3.org/TR/WCAG20/#larger-scaledef) defined in `pt` adds an extra layer of confusion for authors who would normally not actually use this unit of measurement in their CSS/HTML, and then may get very confused about whether or not this refers to actual physical `pt` as measured on the screen (which is something authors have no control over, as it will vary depending on computer/mobile phone/tablet, the OS, the browser, user settings, etc), and whether or not they need to take into account potential variances in dpi (again, not something they have any control over directly).

In short, suggest - in an update/errata to WCAG 2.0 - adding a note about the fact that this refers to CSS `pt` which are anchored on the reference `px` (which then further makes the assumption that the OS/browser/user has a reasonably close mapping of 1px to the idealised reference pixel as per the spec), or simply...using `px` instead (replacing `18pt` with `24px`, `14pt` with `18.5px`, etc)

At a stretch, at least complementing `18pt` with the equivalent `24px` - (e.g. "18pt (24px) ...")

/cc @yatil

**Comment by WayneEDick** [comment]
The point is a physical measure. There is some disagreement on how much of an inch is occupied by a point. I use the Adobe definition that is 1/72 inch. The key is it is not relative. It is a size in inches.

That is the real problem with the pixel. Pixels cannot be used to measure font-size from the point of view of accessibility because the same measure 24px can be from 16 point to 22 point depending on the dpi. 16 point really isn't large print. 

The user does not care how many pixels it takes to make their critical print size, they just want that size because it facilitates reading. For example, to become readable at a 3:1 ratio rather than 4.5:1 requires a physical size in points, inches or centimeters.

I do agree that the point may be a bad measure because its definition is not set. The inch or centimeter would be better.

**Comment by patrickhlauke** [comment]
> The point is a physical measure. There is some disagreement on how much of an inch is occupied by a point. I use the Adobe definition that is 1/72 inch. The key is it is not relative. It is a size in inches.

sorry wayne, but that's not (currently, and for the past few years) correct when talking about web content. from the spec

> For a CSS device, these dimensions are anchored either
>
> (i) by relating the physical units to their physical measurements, or
> (ii) by relating the pixel unit to the reference pixel.

I am not aware of any devices which actually relate any of the physical units (`pt`, `mm`, `cm` etc) that can be used in CSS to actual physical size as measured on screen. All common devices I've tested base their calculation of physical units by referencing its relationship to whatever their 1px size is.

> The user does not care how many pixels it takes...

That may well be, but as a developer has NO actual control over the actual physical size at which, say, a definition of `18pt` is actually rendered (as it depends on a myriad of factors, all outside of the control of the developer), this sentiment cannot be used when defining normative guidelines that developers have to follow.

**Comment by patrickhlauke** [comment]
> That is the real problem with the pixel. Pixels cannot be used to measure font-size from the point of view of accessibility because the same measure 24px can be from 16 point to 22 point depending on the dpi. 16 point really isn't large print.

The problem with using `pt`, `cm`, `mm` etc on the other hand is that their actual physical rendering varies between devices, screen resolutions/dpi, screen dimensions, user settings, browser and OS settings, but many people still assume they actually relate to the real physical size as measured on screen. They don't. Create a webpage, set some container or element to, say, "40pt" width and height, or "2cm" or whatever, then take a ruler out and measure it on a variety of devices - iPad Pro, iPad Mini, a 30" desktop screen set to 1920x1080, etc. You'll find a huge range of actual physical sizes that these supposedly fixed physical units actually render at. This would results in a situation where a developer is being asked/mandated to base a pass/fail of an SC on something that can vary depending on the device they're testing on. And even assuming a developer or auditor has a wide range of devices available and can make a pass/fail evaluation that covers many scenarios, all it takes is for somebody to come along with a device that the original developer/auditor didn't check and come up with a different pass/fail result...which would be ridiculous.

As an aside, this is the same reason why, say, color contrast is measured based on the values indicated in the CSS, rather than saying that it needs to be measured "on screen" with a colorimeter or similar, as again this would make an objective evaluation impossible since monitors/screens will vary in their color rendition based on hardware, color profile, browser, OS, etc.

**Comment by patrickhlauke** [comment]
I guess when this aspect of WCAG 2.0 was originally hashed out, there was still an (optimistic) assumption that browsers/OSs/devices would map `pt` etc to actual physical `pt`. And indeed previous versions of CSS still had this assumption. However, the reality is that the physical units in CSS don't match their actual physical units in the real world (as in most cases, this would require browsers/OSs to take into account physical size of the screen, which they currently don't and in many cases can't). See also the note in https://drafts.csswg.org/css-values/#absolute-lengths

> Note: This definition of the pixel unit and the physical units differs from previous versions of CSS. In particular, in previous versions of CSS the pixel unit and the physical units were not related by a fixed ratio: the physical units were always tied to their physical measurements while the pixel unit would vary to most closely match the reference pixel. (This change was made because too much existing content relies on the assumption of 96dpi, and breaking that assumption broke the content.)

**Comment by patrickhlauke** [comment]
See http://codepen.io/patrickhlauke/pen/zqabMR for a quick demonstration of this - comparing blocks set in `pt` and `px`

**Comment by MichielBijl** [comment]
As far as I know, all OS' and user agents translate `pt`'s into a fixed `px` value. It would thus make sense to just give the information in `px`. That is what a lot of people do anyway, they take the `pt` and simply replace the `t` with an `x` and test against that; which is wrong. This leads to the spec causing harm because it is unclear or at the very least misunderstood.

It would make far more sense to give the font size in `px` than in `pt`, even in an age where we try to get everybody to use `em`'s.

**Comment by yatil** [comment]
That’s actually a hard one to change in general:

WCAG is (or at least tries to be) technology independent. Therefor, I guess `pt` as mentioned in WCAG is actually meant to be the physical size. There even is a note in large scale (text) glossary that says (quote):

> Note 3: The actual size of the character that a user sees is dependent both on the author-defined size and the user's display or user-agent settings. For many mainstream body text fonts, 14 and 18 point is roughly equivalent to 1.2 and 1.5 em or to 120% or 150% of the default size for body text (assuming that the body font is 100%), but authors would need to check this for the particular fonts in use. When fonts are defined in relative units, the actual point size is calculated by the user agent for display. The point size should be obtained from the user agent, or calculated based on font metrics as the user agent does, when evaluating this success criterion. Users who have low vision would be responsible for choosing appropriate settings.

>  but authors would need to check this for the particular fonts in use
concerns me especially, as it implies testing all the devices with the used fonts with an actual ruler to make sure they meet the definition. Which would be ridiculous :-) 

Nowadays the definition should probably be more specific, at least for CSS use, as we have both the Browser font size and the relationship of ´pt´ to pixels defined. Also we should use the “modern” units to ease development:

> In CSS, 14pt equals 18.66px or 1.166rem. 18pt equals 24px or 1.5rem. 

(I think it would be good to have that as a CSS technique somewhere, but I wonder where it would fit.)

**Comment by yatil** [comment]
More info on CSS units: http://www.w3.org/Style/Examples/007/units.en.html by @bert-github, which explicitly says that the `pt`, `cm`, `mm`, `in`, `pc` are not recommended for screen use.

**Comment by StommePoes** [comment]
The number of times the question appears in forums and mailing lists everywhere, every year: "what size is 'large text' according to WCAG?"

It would be great if it were possible to tell people a real answer. Most of us just make up px equivalents and go from there, as an exercise in practicality... which means, if we wanted to write unit/functional tests for a11y, we can't really do that for this particular part, when it should be programmatically testable in, say, Selenium.

**Comment by patrickhlauke** [comment]
> More info on CSS units: http://www.w3.org/Style/Examples/007/units.en.html by @bert-github

note that https://drafts.csswg.org/css-values/#absolute-lengths clarifies the definition that this was based on. 

> This definition of the pixel unit and the physical units differs from previous versions of CSS

**Comment by MichielBijl** [comment]
@yatil 
> (I think it would be good to have that as a CSS technique somewhere, but I wonder where it would fit.)

Two possible places, the [first note on Understanding SC 1.4.3](http://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) and the [definition of “large text”](http://www.w3.org/TR/WCAG20/#larger-scaledef).

**Comment by patrickhlauke** [comment]
For further context, see also when I raised this same issue back in 2014 https://lists.w3.org/Archives/Public/w3c-wai-ig/2014OctDec/0002.html

A key part of that discussion, to me, was Gregg's answer here https://lists.w3.org/Archives/Public/w3c-wai-ig/2014OctDec/0003.html

> The goal is to have text be larger than typical font.    The size that a font appears to the user is dependent on lots of things - like zoom of browser, pixel density of the display, scaling (some displays like the retina) do not render a pixel as one pixel on the display, viewing distance, etc — all of which are outside of the control of the author — so it is meaningless to have an accessibility guideline for web authors that requires that the user see the font in a particular size, or angle subtended.     
>
> So the guidelines were written to talk about minimum RELATIVE size of the font.   If the font is relatively larger — and the user does not use a screen resolution that is too dense — it all works out.    If the user with low vision looks at the content from a distance,  or with a high screen pixel density / screen resolution,  then the author cannot be responsible.     There is an implied contract here — that authors do x part and users do y.   Authors cannot do it all and WCAG doesn’t imply they should.    For example,  Authors do not have to make their content accessible to people who are blind.  They only have to make their content be exposed in a way that a user could use as screen reader to make it accessible to them.   Similarly, for low vision the goal it to put it within reach.  For some - just making it 1.5x font size is sufficient.  For others they may need to zoom the text. 

and https://lists.w3.org/Archives/Public/w3c-wai-ig/2014OctDec/0005.html

> 18/14  point values are based on the fact that the user has a screen/resolution/zoom/viewing distance that makes 1em readable.   
>
> Those point sizes are ONLY used in WCAG to determine when contrast can be reduced.   The assumption is that the text is readable at normal size (due to  the screen/resolution/zoom/viewing distance that the user has set)  and this is just saying that text that is LARGER can have a lower contrast.

and https://lists.w3.org/Archives/Public/w3c-wai-ig/2014OctDec/0007.html

> The comparison should be against a page with no font size specified.     That is,  the “Base font” should be no font size specified.

which all, long story short, actually make me think that the *normative* definition of "Large text" should be modified to actually mention all these additional qualifiers (and it should *not* just be relegated to "we'll fix it in 'Understanding...' or similar").

@yatil 
> WCAG is (or at least tries to be) technology independent. Therefor, I guess pt as mentioned in WCAG is actually meant to be the physical size.

The problem with that is that regardless of technology (HTML/CSS, Flash, PDF, native apps) there is currently no way in any of these to actually define physical "real-world" sizes consistently in a way that is actually guaranteed across devices/browsers/OSs/etc.

Therefore, I'd actually move to change the normative definition of large text to essentially say something more akin to (based on roughly what Gregg said in the 2014 discussion on list)

```
Large scale (text)
text which is at least 1.2 times the size of the base font and bold, or 1.5 times the size of the base font
```

then moving the note about the "actual size" as rendered on an end user's device right after that (and modify *its* wording to then use `px` as example measurements rather than `pt`)

**Comment by yatil** [comment]
@MichielBijl 
I meant a “real” CSS technique, like “CSS3122: Color contrast” under 1.4.3 https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=143#visual-audio-contrast-contrast – But yes, the understanding is the obvious place for such a clarification.

**Comment by patrickhlauke** [comment]
and yes, I agree with @StommePoes and @MichielBijl ... the fact that this discussion/question keeps popping up (and the length of explanation always necessary, even for people like myself who have been working on the web and on accessibility for aeons) is evidence to me that there's something fundamentally borked with the current wording in the normative part of WCAG

**Comment by yatil** [comment]
(Furthermore, back in the day when 16px was humongous, many people basically defined 14pt = 14px and 18pt = 18px, which made some sense as their – chosen – base font was often 10px.)

**Comment by alastc** [comment]
I agree, the place it really needs change is in the definition:
https://www.w3.org/TR/WCAG20/#larger-scaledef 

For years I assume pt=px, so was underestimating the size of large text. 

@yatil Note 3 in the large scale text definition is meaningless, it is trying to compare pt, em & %, when they are all based on (CSS) px. 

A challenge to the group: What device that can show web-content does _not_ base it's sizing on CSS pixels?

**Comment by patrickhlauke** [comment]
and as an aside, just to muddle the water: once authors start to look at WCAG in the context of native (mobile) app development, there's further confusion about what a `pt` means. Apple uses `pt` in their iOS guidelines, but it's not a `pt` in the physical real-world sense, but a shorthand for Apple's own device/density-independent measurement. Android uses device/density independent pixels `dps`. Windows development uses `effective pixels` (again, a device/density independent pixel). It all just gets very confusing (particularly with Apple's reappropriation of the `points` term to mean once again something slightly different from actual physical `pt`)

**Comment by patrickhlauke** [comment]
@alastc
> A challenge to the group: What device that can show web-content does not base it's sizing on CSS pixels?

My money's on "none, or if they do, it's by accident/chance" (using my own http://codepen.io/patrickhlauke/full/zqabMR/ I found that my specific monitor - a 23" Dell at 1920x1080, with Windows running at 1920x1080 as well - actually rendered `72pt` in CSS close enough to a real-world `1in` as measured on screen...but then when I open the exact same thing on my laptop, the measurement is off by a considerable amount, not to mention when you then open it on an iPhone 6s or similar, where `72pt` is closer to `0.5in` as measured on screen)

**Comment by MichielBijl** [comment]
Problem there is that it's silly to specify a value and unit without viewer distance to screen. We live in a world where we a legion of different devices, all of which we hold at different distances from our face. My laptop is generally closer to my face than my iMac. My iPhone even more so. I get that there is need for `pt` as a unit for print etc. There should, however, be an equivalent for the web in `px`.

**Comment by patrickhlauke** [comment]
> Problem there is that it's silly to specify a value and unit without viewer distance to screen.

The assumption has to also be made that the `base font size` is that which the device/OS/browser/etc manufacturer has deemed to be appropriate for their device, based on their device's typical viewing distance. But yes, this is fundamentally *another* variable which makes physical "real-world" size even more ambiguous. Using `px` (under the assumption above that a device is using its "ideal viewport" http://patrickhlauke.github.io/web-tv/ideal-viewport/ / http://www.quirksmode.org/mobile/metaviewport/) at least removes any further ambiguity, I'd say.

**Comment by MakotoUeki** [comment]
> Which is part of my point (hah, no pun intended): as for an author there is no way to actually set anything in CSS and guarantee that it will consistently be presented to a particular physical real-world size, having normative text that actually specifies real-world physical sizes is like asking developers to come up with a magic unicorn technique...

+1

Steve also pointed out this issue at:
https://www.paciellogroup.com/blog/2012/05/whats-large-text-in-wcag-2-0-parlance/

**Comment by cheshrkat** [comment]
All the options are imperfect... but of all the imperfect options, I would prefer to see this information using `px` than `pt` (or list px *as well* assuming the basic conversion metioned earlier in the thread). It's just easier to understand and more likely to result in action.

**Comment by alastc** [comment]
I edited the definition to see how easy it was. I think it was fairly straightforward (although obviously it would have to be a change to normative text).

**large scale (text)**
with at least 24 or 19 pixel (px) bold, or font size that would yield equivalent size for Chinese, Japanese and Korean (CJK) fonts

Note 1: Fonts with extraordinarily thin strokes or unusual features and characteristics that reduce the familiarity of their letter forms are harder to read, especially at lower contrast levels.

Note 2: Font size is the size when the content is delivered. It does not include resizing that may be done by a user.

Note 3: The actual size of the character that a user sees is dependent both on the author-defined size and the user's display or user-agent settings. For many mainstream body text fonts, 19px and 24px is equivalent to 1.2 and 1.5 em or to 120% or 150% of the default size for body text (assuming that the body font is 16px), but authors would need to check this for the particular fonts in use. The pixel size should be obtained from the user agent, or calculated based on font metrics as the user agent does, when evaluating this success criterion. Users who have low vision would be responsible for choosing appropriate settings.

Note 4: When using text without specifying the font size, the smallest font size used on major browsers for unspecified text would be a reasonable size to assume for the font. If a level 1 heading is rendered in 19px bold or higher on major browsers, then it would be reasonable to assume it is large text. Relative scaling can be calculated from the default sizes in a similar fashion.

Note 5: For other fonts such as CJK languages, the "equivalent" sizes would be the minimum large print size used for those languages and the next larger standard large print size.

**Comment by jnurthen** [comment]
This edit would lead to any code which was written to use 14pt bold to no longer be classified as large text. This cannot happen. What was large text before any change must remain large text after any change.

**Comment by alastc** [comment]
How so? 14pt = 19px.

**Comment by MichielBijl** [comment]
@jnurthen this wouldn't change anything. It would just clarify what 14pt bold actually is. `14pt` bold is not the same as `14px` bold.

**Comment by patrickhlauke** [comment]
@alastc 
> I edited the definition to see how easy it was. I think it was fairly straightforward

I'd actually keep it even more high-level at first, as per my proposed amendment

```
Large scale (text)
text which is at least 1.2 times the size of the base font and bold, or 1.5 times the size of the base font
```

and then perhaps add something - a note? - about "for instance, with a base font size of 16px - the common size in most current browsers - this would translate to 19px bold / 24px" or similar

**Comment by jnurthen** [comment]
14pt = 18.666px 
Not 19px. If I define my font as 14pt then I no longer meet this new definition but I did before.

**Comment by patrickhlauke** [comment]
> 14pt = 18.666px

round it down then to the nearest half pixel, I'd say, for sanity. call it 18.5px. but that's a tiny detail in the larger scheme, I'd say (unless we want to argue the tiny fractional values)

**Comment by MichielBijl** [comment]
@jnurthen 18.666px does not exist, you can't have fragmented pixels. With the possible exception of half pixels on 2x density screens.

**Comment by patrickhlauke** [comment]
> @jnurthen 18.666px does not exist, you can't have fragmented pixels. With the possible exception of half pixels on 2x density screens.

not just 2x, but any high-density screens. so the point is indeed valid, but I'd say it's a minor point. not wanting to lose sight of the more fundamental discussion here about `pt` not being the best measure, and needing the clarification

**Comment by jnurthen** [comment]
What the browser does with a fractional pixel is the browsers issue. However, it will report fractional pixels when you query the size of an object. 
Checking  tools use this to determine of something is large font. If you change the definition then code which met the standard before will fail in checking tools. To me this is not acceptable. 

18.5 px is fine.

**Comment by MichielBijl** [comment]
@patrickhlauke yeah that is true. Let's discuss the larger issue :)

**Comment by alastc** [comment]
@jnurthen Last time I tested this, browsers rounded 14pt up to 19px, but I think rounding down in WCAG to 18px would be fine, if needed (considering the next point).

@patrickhlauke Agree with a proportional approach, but rather than talking about the base font-size (which the author can manipulate), should we talk about default user-agent font size?

**Comment by patrickhlauke** [comment]
@alastc well, in the email discussion itself Gregg did talk about "base font size before any sizing is applied". but i actually think that i'm tending more towards base font size in the sense of whatever the author has set as "normal" (as opposed to large) text, due to the note in that same discussion that the sizing here isn't about readability, but proportion. i.e. if an author sets a smaller base font size, users will likely zoom in. then again, not overly married to that, so i could live with "default user-agent font size" or words to that effect (provided this is tech-agnostic enough)

**Comment by patrickhlauke** [comment]
if we're worried about the `19px vs 18.5px` thing, then the scaling factor would be closer to 1.16 instead of 1.2. that's probably still acceptable/nice-enough looking

**Comment by patrickhlauke** [comment]
however, if we're thinking of this for a (versioned, whatever) WCAG.next, then we *could* make previously conformant sizes non-conformant, and stick with the nicer 1.2...

**Comment by alastc** [comment]
quoting & correcting myself: "Last time I tested this, browsers rounded 14pt up to 19px"
That is out of date - now browsers (FF at least) do report the fractions, even on a 1x screen. 
14pt = 18.667px

**Comment by goetsu** [comment]
For history it's something I already mention in 2008 (second part of my comment suggesting to use px)
https://lists.w3.org/Archives/Public/public-comments-wcag20/2008Mar/0088.html who lead to this respond http://trace.wisc.edu/bugzilla_wcag/issuereports/issue_ind.php?id=2430

the other problem is that most people doesn't understand that the large text isn't a fixed value it depend on author choosen size of what is standard text and that's also a point that I think wcag must clarify

**Comment by aardrian** [comment]
@alastc 

> A challenge to the group: What device that can show web-content does not base it's sizing of PT on CSS pixels?

Printers.

Though that may have changed in print drivers and how the browsers hand that stuff off, as it's been a while since I last tested.

**Comment by alastc** [comment]
@aardrian Perhaps, I think it needs testing, but also: printers might  can have separate CSS, and have different sizing. Do you think WCAG needs to cover print? I can't think of any instances where print has been relevant.

**Comment by patrickhlauke** [comment]
@aadrian and even on print, it's mostly dependent on actual printer driver, settings (e.g. if the user choose to "fit to paper" or similar), etc. see for instance my test page, printed with default settings from Chrome. 72pt are rendered larger than the nominal 1in (which is fine of course, but indicates it's all a bit of an approximation there too, so not reliable "absolute" "real-world" physical size either) https://twitter.com/patrick_h_lauke/status/723183050792820736

**Comment by lauracarlson** [comment]
@aadrian @patrickhlauke The Low Vision Task Force has a section on [Printing Customized Text](http://w3c.github.io/low-vision-a11y-tf/requirements.html#printing-customized-text) in the Accessibility Requirements for People with Low Vision doc.

**Comment by patrickhlauke** [comment]
@lauracarlson super. note though that this will be orthogonal to the discussion here about the definition for the purpose of determining color contrast, and i don't think anything discussed here will invalidate/go against any of what I see over on http://w3c.github.io/low-vision-a11y-tf/requirements.html#printing-customized-text

**Comment by lauracarlson** [comment]
@patrickhlauke Sweet!

**Comment by aardrian** [comment]
@patrickhlauke @alastc I agree it is orthogonal. And not just because I like that word.

**Comment by awkawk** [comment]
Good discussion.  

I don't think that we can consider this an errata item as the use of points in WCAG 2.0 may be regarded to not be the best decision but we can't say that it is an error. 

We can add information to the understanding document to make this more clear, but as James indicates we need to make sure that we are making the clarification meet the actual text of the SC.

If we are talking about a change for a future version of WCAG, we may have lots of options, including possibly removing the concept of large text altogether.  I'm not sure what that might look like yet, but we can work on that.

We should think about how this discussion gets resolved in the short-term, whether that means defining changes for the understanding document or adding a WCAG.next label to this issue for future reference, or both.

**Comment by patrickhlauke** [comment]
> I don't think that we can consider this an errata item as the use of points in WCAG 2.0 may be regarded to not be the best decision but we can't say that it is an error.

so "errata" are strictly for errors only, not "it was a mistake perhaps to use `pt`, so we've converted to `px` in the normative definition instead" ?

> we need to make sure that we are making the clarification meet the actual text of the SC.

so the converted `px` value needs to be as close to the original `pt` value, so that if some author WAS using `pt` they could reconvert the `px` value to `pt` and arrive at the same value as before?

> We should think about how this discussion gets resolved in the short-term, whether that means defining changes for the understanding document or adding a WCAG.next label to this issue for future reference, or both.

as the problematic text (use of `pt`) is in the normative definition/glossary of the actual core WCAG 2.0 document, I feel it would be strange and roundabout to then add clarifications/changes to non-normative "understanding" document only - though if that's the only thing we can hope for at this stage, then making it quite prominent and clear (using the various bits from this thread to explain that `pt` is not an actual physical measure, and that it effectively means "1pt = 1.333px") would probably help a bit...

**Comment by awkawk** [comment]
Patrick,
Well, there's substantive and editorial errata, so we might make the case that this is a substantive errata but we can't make the case that this was just a simple mistake like using the wrong language code for an example in WCAG. If it was a substantive errata, we would at this point be dealing with that in a WCAG.next most likely.

For any conversion value, the concern is that if a user follows the advice that the advice meets the normative text.  I do understand from the discussion that this is a particular challenge with the variety of pixel densities and device-specific concerns, so I'm sure that threading that needle is a challenge.

If you think that would help and want to put in a pull request that will help make it happen.

**Comment by yatil** [comment]
For reference, I have used my mini-device lab to test how large a 72pt square actually is on different devices ([test code on codepen](http://codepen.io/yatil/pen/mPGBJE)) – and on different screens on the same device. Here are my findings ([Flickr album](https://www.flickr.com/photos/yatil/albums/72157667533036426)):

* [MacBook Air 13" Non-Retina](https://www.flickr.com/photos/yatil/26625862746/in/album-72157667533036426/): 72pt in CSS are rendered as 2cm (.787 inches).
  * [attached old-ish external monitor](https://www.flickr.com/photos/yatil/26652612785/in/album-72157667533036426/): 72pt in CSS are rendered as about 2.5cm (about 1 inch).
* [Amazon Kindle Paperwhite](https://www.flickr.com/photos/yatil/26625864276/in/album-72157667533036426/): 72pt in CSS are rendered as about 2.5cm (about 1 inch).
* [ZTE Firefox OS Phone](https://www.flickr.com/photos/yatil/26625865456/in/album-72157667533036426/): 72pt in CSS are rendered as about 1.65cm (about 0.65 inches).
* [iPhone 6s Plus – Standard mode](https://www.flickr.com/photos/yatil/26652614915/in/album-72157667533036426/): 72pt in CSS are rendered as about 1.7cm (about 0.67 inches).
* [iPhone 6s Plus – Zoom mode](https://www.flickr.com/photos/yatil/26046648324/in/album-72157667533036426/): 72pt in CSS are rendered as about 1.8cm (about 0.71 inches).
* [Printed on paper](https://www.flickr.com/photos/yatil/26652702485/in/album-72157667533036426/): 72pt in CSS are rendered as 2.54cm (1inch).

There is no way to make sure that a web author can make sure that text is rendered at a specific physical size on devices, and even different screens on the same device might differ in resolution. It makes sense to define a relationship to the base text. This makes sure that a low(er)-contrast text is at least – for example – 1.2⨉ larger than the base font. In addition, we might want to consider to introduce a minimum font size in WCAG.next. That could for example be 16px (today’s browser default) or 18.66px (aka 14 CSS-pt). 

_(Note that this was not done using any high precision device but a ruler, but the differences are large enough to make the point. – Also I now know why German craftspeople like to give out free rulers with their company name on it: Internet fame.)_

**Comment by patrickhlauke** [comment]
The elephant in the room, as far as what large text means in the context of contrast measurement, is also: even assuming that WCAG 2.0 meant "actual size as rendered on screen", and even assuming that there was some magical, not yet invented magic pony technology that did allow authors to define exactly how their content would be rendered in physical screen sizes...this would still not take into account (average or actual) viewing distance, which is yet another factor that determines how big text is actually perceived by the user, which in turn is a factor in determining what minimum contrast would be acceptable.

In short, the entire premise, no matter how you slice it, is built on too many variables that are completely subjective/variable.

In order to actually make this testable and have consistent results, WCAG 2.0 *must* (IMO):

a) clarify that it means a unit of measure that can be programmatically queried (be it "px", or "pt", or whatever...but it needs to clearly say that it doesn't mean "as measured on the screen with a ruler")
b) define basic anchoring parameters, as otherwise you're dealing with too many variables - the most reliable anchoring would be the UA's default font size, under the situation where - on mobile/tablet - the ideal viewport (using `width=device-width` in viewport meta) is used.

**Comment by dstorey** [comment]
Something that shows pt being mapped to logical CSS pixels in CSS is if you compare the size of PTs in an app vs a web browser. In this case Word (100% zoom / actual size). There is a huge difference, which is one of the things that leads to designer/dev confusion when moving from mocks to implementation

![screen shot 2016-04-26 at 10 16 03 pm](https://cloud.githubusercontent.com/assets/465069/14842486/abfe42be-0bfc-11e6-8c4f-51377ab1a715.png)

Also our Microsoft data (1,218,301 unique URLs crawled using Edge and Chrome) clearly shows that PX is way more widely used: https://developer.microsoft.com/en-us/microsoft-edge/platform/usage/css/font-size/

PT is used less than Ems and %. Also bear in mind that this includes all rules that applies, so I believe print styles will be picked up here. It also doesn't show how much more a unit is used on a site, just if it is used once or not. I suspect on the sites where both are used, PX is used in a lot more declarations.

**Comment by patrickhlauke** [comment]
Proposed changes to at least the notes part of the glossary definition: https://github.com/w3c/wcag/pull/184

**Comment by awkawk** [comment]
@patrickhlauke The pull request for this changes the core WCAG 2.0 document.  Do you want to suggest changes to the understanding document or do you want to mark this with the WCAG next label and close it until we are ready to consider that?

**Comment by patrickhlauke** [comment]
@awkawk so to be clear: can nothing in WCAG 2.0 be changed (anymore), not even informative/non-normative notes? Is this not essentially an editorial clarification?

i believe having this shuffled off to understanding may not work out (particularly if in future other SCs may crop up that do need to refer to the definition of what "large scale (text)" is - which would then need this to be duplicated/triplicated in each relevant SC's understanding)

If all else fails, then I guess the only option would be WCAG next...

**Comment by awkawk** [comment]
I don't think that this is editorial.  If you would like to ask the group we can, but I'd rather focus our energies on the wag next path.

**Comment by patrickhlauke** [comment]
I would like to ask the group one time directly if this can be considered editorial, thus ok for inclusion in WCAG 2.0 - maybe as part of a survey? After that, if the group does not see it as editorial, happy (of sorts) to slate this for the next path.

**Comment by mraccess77** [comment]
IMO the lessening of the contrast requirement should be in relationship to size of the default font and shouldn't really be specified in pt or px.  I agree that it's no guarantee it's readable and many people with low vision need larger fonts and that if you go larger it doesn't mean the need for contrast goes away.    I believe the goal of the criteria was to lessen the level of contrast for text that was 1.2x bold or 1.5 the size.  Since many sites are now using mobile we can't really assume 12pt/16px as default font size and using this is problematic.  These units should only be examples to the requirement which is measured by 1.2x bold or 1.5 the default font size.

**Comment by alastc** [comment]
@mraccess77 That is pretty much what the explanation says and directs people to do, more clearly than the original text (IMHO). 

The first question is whether we can change the notes under the definition (is it editorial or substantive), which is on the survey. Then we can get onto the exact wording.

**Comment by mraccess77** [comment]
@alastc Yes, just lending my support to the push for change.  Regarding whether it can be changed or not -- while I think we could change a note in a definition section I don't think we can change the definition part that is not preceded by a note.  But this is not my area to comment on.

**Comment by alastc** [comment]
In this case it would change almost all of the 'notes' under the definition, which are fairly lengthy (5 or 6 paragraphs), but not the definition itself.

**Comment by patrickhlauke** [comment]
> Since many sites are now using mobile we can't really assume 12pt/16px as default font size and using this is problematic

even on mobile, the default font size is 16 CSS px on most common devices. how big those CSS px actually appear will then depend on various factors, including whether or not the site was set to use an ideal viewport of `width=device-width` and whether the user has zoomed in/out...but those are factors that we can't take into account (nor detect), so they are irrelevant (for the context of color contrast, not readability)

**Comment by yatil** [comment]
Just for consideration: Making the distinction between large and small text by setting a relationship is useful with normal-sized text but gets ridiculous when someone is using a large base text size (like 25px). Should they be not allowed to use the 3:1 ratio? I think in the future we need to find another measurement for this – or get rid of the exception altogether.

**Comment by patrickhlauke** [comment]
btw, for clarity...as i see a few comments flying in on this - would be good to get feedback on the actual hard proposal here https://github.com/w3c/wcag/pull/184 (which spawned from this discussion and the discussion on list)

**Comment by joshueoconnor** [comment]
@patrickhlauke Thanks for your work on this - very interesting. The consensus was that we will not incorporate the suggested changes into WCAG 2.0 but work on updating the Understanding document and consider this issue flagged for the WCAG 2.1 release. You can follow the discussion in the minutes.[1]

[1] http://www.w3.org/2016/05/17-wai-wcag-minutes.html

**Comment by patrickhlauke** [comment]
@joshueoconnor and which "Understanding" would that be exactly? as this is in the general definitions, and not in a particular SC's prose, it doesn't naturally belong in any particular understanding doc (though in practice that definition IS only referenced by the color contrast SC)

**Comment by patrickhlauke** [comment]
if this is being deferred to next, i think it would actually make matters a lot clearer (as the situation around differing device sizes, zoom levels, viewports, etc is only bound to become MORE confusing, rather than less, in future) to seriously consider what @yatil mentions...

> or get rid of the exception altogether

Set a single contrast ratio, no exceptions.

**Comment by alastc** [comment]
I've created an update to the contrast understanding doc:
https://github.com/w3c/wcag/commit/dbaf3d0336cc2cad1fd954b6792b62b07487ff14?diff=split

But I'm struggling to work out how to create a separate pull request from the previous one, when I start that process it seems to combine it on top of the previous one, is that ok?

**Comment by alastc** [comment]
With a little git-foo I've created a new branch and got a pull request for updating two understanding docs:
https://github.com/w3c/wcag/pull/188

**Comment by patrickhlauke** [comment]
Closing this, as per discussion/advice on process for these deferred items.
